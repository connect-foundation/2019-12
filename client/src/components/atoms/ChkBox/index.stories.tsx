import React from 'react';

import ChkBox from '.';

export default {
  title: 'Atoms / ChkBox',
};

export const index: React.FC = () => {
  return <ChkBox checked={false} />;
};
