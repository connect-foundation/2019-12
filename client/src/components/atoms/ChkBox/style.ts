import styled from 'styled-components';
import { palette } from 'styled-tools';
import { Icon } from 'components';

export const Wrapper = styled.div`
  width: 2.5rem;
  height: 2.5rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: ${palette('white')};
  border-radius: 0.2rem;
  border: 0.1rem solid ${palette('grayscale', 3)};

  &:hover {
    background-color: ${palette('grayscale', 6)};
  }
`;

export const ChkIcon = styled(Icon)`
  height: 1.8rem;
  user-select: none;
`;
