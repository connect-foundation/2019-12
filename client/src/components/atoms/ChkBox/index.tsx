import React, { useState } from 'react';

import * as S from './style';
import CheckSvg from 'assets/img/check.svg';

export interface Props {
  checked: boolean;
}

function ChkBox({ checked }: Props): React.ReactElement {
  const [isChecked, setIsChecked] = useState(checked);

  return (
    <S.Wrapper
      onClick={() => {
        setIsChecked(!isChecked);
      }}
    >
      {isChecked && <S.ChkIcon alt={'check'} src={CheckSvg} />}
    </S.Wrapper>
  );
}

export default ChkBox;
