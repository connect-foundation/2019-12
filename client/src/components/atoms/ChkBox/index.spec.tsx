import React from 'react';
import { mount } from 'enzyme';
import ChkBox from '.';

describe('Atom / ChkBox', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(<ChkBox checked={false} />);

    // then
    expect(wrapper).toMatchSnapshot();
  });

  it('[PROP] checked에 따라 체크 표시가 된다.', () => {
    // given
    const wrapper = mount(<ChkBox checked={false} />);
    const wrapperChcked = mount(<ChkBox checked={true} />);

    // then
    const imgElement = wrapper.find('img');
    const imgElementBeShown = wrapperChcked.find('img');

    expect(imgElement).toHaveLength(0);
    expect(imgElementBeShown).toHaveLength(1);
  });

  it('체크박스를 클릭했을 때 체크 표시된다.', () => {
    // given
    const wrapper = mount(<ChkBox checked={false} />);

    // when
    wrapper.simulate('click');

    // then
    const imgElement = wrapper.find('img');
    expect(imgElement).toHaveLength(1);
  });
});
