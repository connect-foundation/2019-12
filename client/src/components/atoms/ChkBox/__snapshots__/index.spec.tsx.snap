// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / ChkBox [SNAPSHOT] 렌더링 1`] = `
.c0 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.2rem;
  border: 0.1rem solid;
}

<ChkBox
  checked={false}
>
  <styled.div
    data-testid="ticketbox-chkbox"
    onClick={[Function]}
  >
    <StyledComponent
      data-testid="ticketbox-chkbox"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwCtUz",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 2.5rem;
  height: 2.5rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: ",
              [Function],
              ";
  border-radius: 0.2rem;
  border: 0.1rem solid ",
              [Function],
              ";

  &:hover {
    background-color: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwCtUz",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <div
        className="c0"
        data-testid="ticketbox-chkbox"
        onClick={[Function]}
      />
    </StyledComponent>
  </styled.div>
</ChkBox>
`;
